wallet/wallet.c

```diff
@@ -1609,6 +1610,12 @@ void wallet_payment_store(struct wallet *wallet,
 					    payment->route_channels);
 	sqlite3_bind_int64(stmt, 9, payment->msatoshi_sent);
 
+	if (payment->description != NULL)
+		sqlite3_bind_blob(stmt, 10, payment->description,
+				  tal_len(payment->description), SQLITE_TRANSIENT);
```

Comment:
Don't use tal_len() here, as it's not well-defined for tal_strdup().  This works only because in developer mode we define CCAN_TAL_DEBUG and that *always* assigns a length when it allocates.  I will fix this today so it correctly asserts if you ask for tal_len/tal_count on a unlengthed object...

Reply:
ie. sqlite3_bind_text()

Reply:
I was not aware that there is this quirk, I was always under the impression that things allocated with `tal_*` or `tal` would be tagged with their length and would therefor be acceptable for `tal_len`.

Reply:
Yeah :( That's why I fixed it to be that way in my other PR. "If the guy with the PhD can't use your interface, maybe your interface sucks" :)